Aos 2 boundary objects efic

efifogel opened the pull request:
## Summary of Changes

Two-dimensional arrangement of curves embedded in an orientable parametric surface in three dimensional space.

## Release Management

* Affected package(s): Arrangement_on_surface_2, Boolean_set_operations_2
* Issue(s) solved (if any): None
* Feature/Small Feature (if any): https://cgal.geometryfactory.com/CGAL/Members/wiki/Features/Arrangement_on_surface_2
* Link to compiled documentation [2D Arrangements](http://www.cs.tau.ac.il/~efif/doc_output2/Arrangement_on_surface_2/)
* License and copyright ownership: Tel Aviv Univ.

This feature is under review.
The link to the documentation is for the 2nd and latest submission at the time this pull-request has been created.

maxGimeno commented:
There are conflicts with master

efifogel commented:
I've fixed the conflicts, but we are waiting for Michale Hofmann's review I
guess.
   ____  _        ____             _
  /_____/_) o    /__________  __  //
 (____ (   (    (    (_/ (_/-(-'_(/
                         _/




On Mon, 22 Feb 2021 at 10:15, Maxime GIMENO <notifications@github.com>
wrote:

> There are conflicts with master
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/CGAL/cgal/pull/5223#issuecomment-783182345>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABVBNOBURCYQI4NJVTEITVTTAIHANANCNFSM4UHL444A>
> .
>

lrineau commented:
> I've fixed the conflicts

The merge with the fixed conflicts does not appear in the history. Maybe you fixed on your machine but did not push.

efifogel commented:
As usual, I used an old branch linked to cgal-public-dev/ instead of mine/.
I'll fix it shortly.
   ____  _        ____             _
  /_____/_) o    /__________  __  //
 (____ (   (    (    (_/ (_/-(-'_(/
                         _/




On Mon, 1 Mar 2021 at 17:03, Laurent Rineau <notifications@github.com>
wrote:

> I've fixed the conflicts
>
> The merge with the fixed conflicts does not appear in the history. Maybe
> you fixed on your machine but did not push.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/CGAL/cgal/pull/5223#issuecomment-788016841>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABVBNOEBN7XDAKX2HFI3GQLTBOUDXANCNFSM4UHL444A>
> .
>

maxGimeno commented:
There are tabs in here : Arrangement_on_surface_2/doc/Arrangement_on_surface_2/Arrangement_on_surface_2.txt

efifogel commented:
I've removed them.
   ____  _        ____             _
  /_____/_) o    /__________  __  //
 (____ (   (    (    (_/ (_/-(-'_(/
                         _/




On Tue, 2 Mar 2021 at 09:38, Maxime GIMENO <notifications@github.com> wrote:

> There are tabs in here :
> Arrangement_on_surface_2/doc/Arrangement_on_surface_2/Arrangement_on_surface_2.txt
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/CGAL/cgal/pull/5223#issuecomment-788690223>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABVBNOEOXYTHI2FHG2YQXZTTBSIVXANCNFSM4UHL444A>
> .
>

maxGimeno commented:
Thank you, but now it is `Arrangement_on_surface_2/examples/Arrangement_on_surface_2/arr_rat_functions.h` that has trailing whitespaces :/

efifogel commented:
Fixed
   ____  _        ____             _
  /_____/_) o    /__________  __  //
 (____ (   (    (    (_/ (_/-(-'_(/
                         _/




On Tue, 2 Mar 2021 at 12:51, Maxime GIMENO <notifications@github.com> wrote:

> Thank you, but now it is
> Arrangement_on_surface_2/examples/Arrangement_on_surface_2/arr_rat_functions.h
> that has trailing whitespaces :/
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/CGAL/cgal/pull/5223#issuecomment-788816449>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABVBNOD4MO3JH7TEUAIZBXDTBS7KBANCNFSM4UHL444A>
> .
>

maxGimeno commented:
The CI now says : `/home/runner/work/cgal/cgal/Installation/../Arrangement_on_surface_2/include/CGAL/Arr_spherical_gaussian_map_3/Arr_polyhedral_sgm.h:98:20: error: ‘cross_product’ is not a member of ‘CGAL’`

lrineau commented:
> The CI now says : `/home/runner/work/cgal/cgal/Installation/../Arrangement_on_surface_2/include/CGAL/Arr_spherical_gaussian_map_3/Arr_polyhedral_sgm.h:98:20: error: ‘cross_product’ is not a member of ‘CGAL’`

That function is declared and defined in `<CGAL/Kernel/global_functions_3.h>`.

maxGimeno commented:
the incldue is missing, then

efifogel commented:
I've added it, but how come I don't get the error when I run the tests?
   ____  _        ____             _
  /_____/_) o    /__________  __  //
 (____ (   (    (    (_/ (_/-(-'_(/
                         _/




On Wed, 3 Mar 2021 at 10:06, Maxime GIMENO <notifications@github.com> wrote:

> the incldue is missing, then
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/CGAL/cgal/pull/5223#issuecomment-789522341>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABVBNOFRUSKXB2J3S2LUXGTTBXUXHANCNFSM4UHL444A>
> .
>

maxGimeno commented:
In HEADER_ONLY, most of the missing includes are still found for some reason, but the checks from the CI must do something a little special and detect it better.

efifogel commented:
ok, no problem;
keep feeding me until all issues are resolved.
   ____  _        ____             _
  /_____/_) o    /__________  __  //
 (____ (   (    (    (_/ (_/-(-'_(/
                         _/




On Wed, 3 Mar 2021 at 11:09, Maxime GIMENO <notifications@github.com> wrote:

> In HEADER_ONLY, most of the missing includes are still found for some
> reason, but the checks from the CI must do something a little special and
> detect it better.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/CGAL/cgal/pull/5223#issuecomment-789559237>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABVBNOFW26LREXPYQQEALI3TBX4F3ANCNFSM4UHL444A>
> .
>

maxGimeno commented:
Apparently there are new dependencies to AOS2 : 
```
Modifier
Polyhedron
Spatial_sorting
TDS_2
```

If they are expected, they should be added to `Arrangement_on_surface_2/package_info/Arrangement_on_surface_2/dependencies`

efifogel commented:
ok, that triggers a potential commercial problem.
let's first put order into these dependencies.

   1. Modifier
   2. Polyhedron
   3. Spatial_sorting
   4. TDS_2


   - (4) Has been there for a while. It is used for a point location
   strategy based on triangulation, and is mainly there for testing. I think
   that we need a mechanism to make a dependency conditional. Naturally, users
   should be able to use the Arrangement_on_surface_2 package  package without
   the TDS_2 at all.
   - (1) and (2) have been introduced by this new feature. In particular,
   there are needed for whatever goes under
   Arrangement_on_surface_2/include/CGAL/Arr_spherical_gaussian_map_3. This
   should be separated into a new package (because of several reasons). Notice
   that the code has been lying there for quite a while, so in fact the
   dependency is not new. After the feature is accepted, I will introduce a
   new package and move the relevant code to the new package.
   - Can you easily tell me where is Spatial_sorting used?

   ____  _        ____             _
  /_____/_) o    /__________  __  //
 (____ (   (    (    (_/ (_/-(-'_(/
                         _/




On Wed, 3 Mar 2021 at 11:58, Maxime GIMENO <notifications@github.com> wrote:

> Apparently there are new dependencies to AOS2 :
>
> Modifier
> Polyhedron
> Spatial_sorting
> TDS_2
>
> If they are expected, they should be added to
> Arrangement_on_surface_2/package_info/Arrangement_on_surface_2/dependencies
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/CGAL/cgal/pull/5223#issuecomment-789590352>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABVBNOBVGY2EURNZP4XIY7DTBYB37ANCNFSM4UHL444A>
> .
>

lrineau commented:
> In HEADER_ONLY, most of the missing includes are still found for some reason, but the checks from the CI must do something a little special and detect it better.

The CI tests several things that the testsuite does not test. In particular, it runs a long process to check that all documented headers from CGAL are self-contained and can be compiled independently.

maxGimeno commented:
> - Can you easily tell me where is Spatial_sorting used?

I'm afraid not, the diff is far too big and it doesn't seem obvious. I suspect it is somehow related to TDS_2, but I can't be sure.

sloriot commented:
Spatial sorting is used in the insertion by batch in Delaunay based triangulations.

efifogel commented:
I've added the missing 4 dependencies (with a long commit message).
We'll deal with the true-dependency issue at some point in the future.
   ____  _        ____             _
  /_____/_) o    /__________  __  //
 (____ (   (    (    (_/ (_/-(-'_(/
                         _/




On Wed, 3 Mar 2021 at 13:48, Sebastien Loriot <notifications@github.com>
wrote:

> Spatial sorting is used in the insertion by batch in Delaunay based
> triangulations.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/CGAL/cgal/pull/5223#issuecomment-789657939>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABVBNOEKQQU3MZ2JFTAMLTTTBYO2NANCNFSM4UHL444A>
> .
>

maxGimeno commented:
There are conflicts with master

efifogel commented:
Thanks; fixed.
   ____  _        ____             _
  /_____/_) o    /__________  __  //
 (____ (   (    (    (_/ (_/-(-'_(/
                         _/




On Mon, 22 Mar 2021 at 12:08, Maxime GIMENO ***@***.***>
wrote:

> There are conflicts with master
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/CGAL/cgal/pull/5223#issuecomment-803940130>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABVBNOHX5SD7RLMYLZDEFKLTE4JJRANCNFSM4UHL444A>
> .
>

maxGimeno commented:
Thank you

sloriot commented on Arrangement_on_surface_2/doc/Arrangement_on_surface_2/Arrangement_on_surface_2.txt:
```diff
@@ -1810,398 +3494,760 @@ concepts, in turn, refines the concept `ArrangementBasicTraits_2`.
+is equal to \f$x_{\rm min}\f$, \f$x_{\rm max}\f$, or a value in
+between. Return `CGAL::ARR_LEFT_BOUNDARY`, `CGAL::ARR_RIGHT_BOUNDARY`,
+or `CGAL::ARR_INTERIOR`, accordingly. If \f$c\f$ is vertical and lies
+on a vertical boundary; that is, either \f$\forall p=(x_p, y_p) \in c$
```
```suggestion
on a vertical boundary; that is, either \f$\forall p=(x_p, y_p) \in c\f$
```

sloriot commented on Arrangement_on_surface_2/doc/Arrangement_on_surface_2/Arrangement_on_surface_2.txt:
```diff
@@ -1810,398 +3494,760 @@ concepts, in turn, refines the concept `ArrangementBasicTraits_2`.
+between. Return `CGAL::ARR_LEFT_BOUNDARY`, `CGAL::ARR_RIGHT_BOUNDARY`,
+or `CGAL::ARR_INTERIOR`, accordingly. If \f$c\f$ is vertical and lies
+on a vertical boundary; that is, either \f$\forall p=(x_p, y_p) \in c$
+we have $x_p = x_{\rm min}\f$ or \f$\forall p=(x_p, y_p) \in c$ we
```
```suggestion
we have $x_p = x_{\rm min}\f$ or \f$\forall p=(x_p, y_p) \in c\f$ we
```

sloriot commented on Arrangement_on_surface_2/doc/Arrangement_on_surface_2/CGAL/Arr_geodesic_arc_on_sphere_traits_2.h:
```diff
@@ -0,0 +1,453 @@
+       * \param[in] normal the normal to the plane containing the arc.
+       * \pre Both endpoints lie on the given plane.
+       */
+      Curve_2 operator()(const Point_2& p, const Point_2& q,
```
I still don't understand how to construct a curve describe the largest part of the circle between `p` and `q`.

efifogel commented:
In case that wasn't clear, I refer to the oriented plane, so there are two
planes and two corresponding normals associated with the two arcs,
respectively.
   ____  _        ____             _
  /_____/_) o    /__________  __  //
 (____ (   (    (    (_/ (_/-(-'_(/
                         _/




On Wed, 12 May 2021 at 03:36, Efi Fogel ***@***.***> wrote:

> Except for the constructor that accepts  two endpoints., there is a
> constructor that accepts the normal to the plane containing the great arc
> circle as well as the two endpoints. There is also a constructor that
> accepts only the normal that constructs a full great circle.
>
> On Tue, May 11, 2021, 23:56 Sebastien Loriot ***@***.***>
> wrote:
>
>> ***@***.**** commented on this pull request.
>> ------------------------------
>>
>> In
>> Arrangement_on_surface_2/doc/Arrangement_on_surface_2/CGAL/Arr_geodesic_arc_on_sphere_traits_2.h
>> <https://github.com/CGAL/cgal/pull/5223#discussion_r630534958>:
>>
>> > +       *
>> +       * \param[in] p the first endpoint.
>> +       * \param[in] q the second endpoint.
>> +       * \pre p and q must not coincide.
>> +       * \pre p and q cannot be antipodal.
>> +       */
>> +      Curve_2 operator()(const Point_2& p, const Point_2& q);
>> +
>> +      /*! Construct a geodesic arc from two endpoints contained
>> +       * in a plane.
>> +       * \param[in] p the first endpoint.
>> +       * \param[in] q the second endpoint.
>> +       * \param[in] normal the normal to the plane containing the arc.
>> +       * \pre Both endpoints lie on the given plane.
>> +       */
>> +      Curve_2 operator()(const Point_2& p, const Point_2& q,
>>
>> I still don't understand how to construct a curve describing the largest
>> part of the circle between p and q.
>>
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/CGAL/cgal/pull/5223#pullrequestreview-657224952>, or
>> unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/ABVBNOFLP6XMYKVYSUJBBQDTNGKXLANCNFSM4UHL444A>
>> .
>>
>

efifogel commented:
I've enhanced the manual a bit. In particular I've added a figure and some
text that explains in more detail how a minor or a major arc can be
constructed; see bottom of section
http://www.cs.tau.ac.il/~efif/doc_output4/Arrangement_on_surface_2/index.html#arr_ssectr_spherical

I've also added a couple of missing precondition and enhanced the reference
(similar to the above) ; see
http://www.cs.tau.ac.il/~efif/doc_output4/Arrangement_on_surface_2/classCGAL_1_1Arr__geodesic__arc__on__sphere__traits__2.html

   ____  _        ____             _
  /_____/_) o    /__________  __  //
 (____ (   (    (    (_/ (_/-(-'_(/
                         _/




On Wed, 12 May 2021 at 10:22, Efi Fogel ***@***.***> wrote:

> In case that wasn't clear, I refer to the oriented plane, so there are two
> planes and two corresponding normals associated with the two arcs,
> respectively.
>    ____  _        ____             _
>   /_____/_) o    /__________  __  //
>  (____ (   (    (    (_/ (_/-(-'_(/
>                          _/
>
>
>
>
> On Wed, 12 May 2021 at 03:36, Efi Fogel ***@***.***> wrote:
>
>> Except for the constructor that accepts  two endpoints., there is a
>> constructor that accepts the normal to the plane containing the great arc
>> circle as well as the two endpoints. There is also a constructor that
>> accepts only the normal that constructs a full great circle.
>>
>> On Tue, May 11, 2021, 23:56 Sebastien Loriot ***@***.***>
>> wrote:
>>
>>> ***@***.**** commented on this pull request.
>>> ------------------------------
>>>
>>> In
>>> Arrangement_on_surface_2/doc/Arrangement_on_surface_2/CGAL/Arr_geodesic_arc_on_sphere_traits_2.h
>>> <https://github.com/CGAL/cgal/pull/5223#discussion_r630534958>:
>>>
>>> > +       *
>>> +       * \param[in] p the first endpoint.
>>> +       * \param[in] q the second endpoint.
>>> +       * \pre p and q must not coincide.
>>> +       * \pre p and q cannot be antipodal.
>>> +       */
>>> +      Curve_2 operator()(const Point_2& p, const Point_2& q);
>>> +
>>> +      /*! Construct a geodesic arc from two endpoints contained
>>> +       * in a plane.
>>> +       * \param[in] p the first endpoint.
>>> +       * \param[in] q the second endpoint.
>>> +       * \param[in] normal the normal to the plane containing the arc.
>>> +       * \pre Both endpoints lie on the given plane.
>>> +       */
>>> +      Curve_2 operator()(const Point_2& p, const Point_2& q,
>>>
>>> I still don't understand how to construct a curve describing the largest
>>> part of the circle between p and q.
>>>
>>> —
>>> You are receiving this because you authored the thread.
>>> Reply to this email directly, view it on GitHub
>>> <https://github.com/CGAL/cgal/pull/5223#pullrequestreview-657224952>,
>>> or unsubscribe
>>> <https://github.com/notifications/unsubscribe-auth/ABVBNOFLP6XMYKVYSUJBBQDTNGKXLANCNFSM4UHL444A>
>>> .
>>>
>>

sloriot commented:
@efifogel looks like there are some conflicts now. The feature has been accepted and should target 5.4. @maxGimeno as soon as Efi has fixed the conflicts + the CI works could you run the small testsuite on this branch?

efifogel commented:
All conflicts have been fixed.
It's ready for integration.
   ____  _        ____             _
  /_____/_) o    /__________  __  //
 (____ (   (    (    (_/ (_/-(-'_(/
                         _/




On Thu, 29 Jul 2021 at 12:04, Sebastien Loriot ***@***.***>
wrote:

> @efifogel <https://github.com/efifogel> looks like there are some
> conflicts now. The feature has been accepted and should target 5.4.
> @maxGimeno <https://github.com/maxGimeno> as soon as Efi has fixed the
> conflicts + the CI works could you run the small testsuite on this branch?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/CGAL/cgal/pull/5223#issuecomment-888941429>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABVBNOACRYOIURWP4XOS5Z3T2EKQPANCNFSM4UHL444A>
> .
>

maxGimeno commented:
/testme

maxGimeno commented:
/testme

maxGimeno commented:
warnings:
 - `CGAL/Arrangement_2/Arr_traits_adaptor_2.h:2055:12: warning: variable 'equal' set but not used `
 - `CGAL/Arrangement_2/Arr_traits_adaptor_2.h:2766:12: warning: variable 'cmp_y_at_x_right' set but not used`
 - `CGAL/Arrangement_2/Arr_traits_adaptor_2.h:2661:38: warning: equality comparison result unused`
errors:
- `Error: cannot open file data/linear/lines/compare_x_at_limit`
- `Error: cannot open file data/linear/lines/compare_x_near_limit`
- `Error: cannot open file data/rational_arcs/compare_x_at_limit`
- `Error: cannot open file data/rational_arcs/compare_x_near_limit`

maxGimeno commented:
/testme

maxGimeno commented:
new warnings : 
 - `CGAL/Arrangement_2/Arr_traits_adaptor_2.h:2056:12: warning: unused variable 'max_vertex' [-Wunused-variable]
      auto max_vertex = m_self->construct_max_vertex_2_object();`
- `CGAL/Arrangement_2/Arr_traits_adaptor_2.h:2055:12: warning: unused variable 'min_vertex' [-Wunused-variable]
      auto min_vertex = m_self->construct_min_vertex_2_object();`
-

maxGimeno commented:
also warnings :
https://cgal.geometryfactory.com/~mgimeno/test_suite/TESTRESULTS/CGAL-5.4-Ic-5223/Arrangement_on_surface_2/TestReport_cgaltest_Ubuntu-gcc7.gz

efifogel commented:
fixed; let's try again.
   ____  _        ____             _
  /_____/_) o    /__________  __  //
 (____ (   (    (    (_/ (_/-(-'_(/
                         _/




On Tue, 10 Aug 2021 at 11:43, Maxime GIMENO ***@***.***>
wrote:

> also warnings :
>
> https://cgal.geometryfactory.com/~mgimeno/test_suite/TESTRESULTS/CGAL-5.4-Ic-5223/Arrangement_on_surface_2/TestReport_cgaltest_Ubuntu-gcc7.gz
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/CGAL/cgal/pull/5223#issuecomment-895845737>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABVBNOGNDKP3C2YOKHFQMKTT4DRDBANCNFSM4UHL444A>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&utm_campaign=notification-email>
> .
>

maxGimeno commented:
Erros on windows : 
 - https://cgal.geometryfactory.com/~mgimeno/test_suite/TESTRESULTS/CGAL-5.4-Ic-5223/Arrangement_on_surface_2/TestReport_cgaltest_VS-19.gz
 - https://cgal.geometryfactory.com/~mgimeno/test_suite/TESTRESULTS/CGAL-5.4-Ic-5223/Boolean_set_operations_2/TestReport_cgaltest_VS-19.gz
 - https://cgal.geometryfactory.com/~mgimeno/test_suite/TESTRESULTS/CGAL-5.4-Ic-5223/Envelope_3/TestReport_cgaltest_VS-19.gz

efifogel commented:
Fixed.
Let's try again.
   ____  _        ____             _
  /_____/_) o    /__________  __  //
 (____ (   (    (    (_/ (_/-(-'_(/
                         _/




On Tue, 10 Aug 2021 at 13:32, Maxime GIMENO ***@***.***>
wrote:

> Erros on windows :
>
>    -
>    https://cgal.geometryfactory.com/~mgimeno/test_suite/TESTRESULTS/CGAL-5.4-Ic-5223/Arrangement_on_surface_2/TestReport_cgaltest_VS-19.gz
>    -
>    https://cgal.geometryfactory.com/~mgimeno/test_suite/TESTRESULTS/CGAL-5.4-Ic-5223/Boolean_set_operations_2/TestReport_cgaltest_VS-19.gz
>    -
>    https://cgal.geometryfactory.com/~mgimeno/test_suite/TESTRESULTS/CGAL-5.4-Ic-5223/Envelope_3/TestReport_cgaltest_VS-19.gz
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/CGAL/cgal/pull/5223#issuecomment-895917709>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABVBNOFVITVHOMYRSZRIAH3T4D52HANCNFSM4UHL444A>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&utm_campaign=notification-email>
> .
>

maxGimeno commented:
/testme

efifogel commented:
Fix a few more warnings
   ____  _        ____             _
  /_____/_) o    /__________  __  //
 (____ (   (    (    (_/ (_/-(-'_(/
                         _/




On Thu, 12 Aug 2021 at 13:08, Maxime GIMENO ***@***.***>
wrote:

> There are still warnings and errors, but only in Arrangement examples:
> https://cgal.geometryfactory.com/~mgimeno/test_suite/TESTRESULTS/index.shtml#Arrangement_on_surface_2_Examples
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/CGAL/cgal/pull/5223#issuecomment-897512655>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABVBNOH6DJGHSYGGNXDE7JTT4OMRPANCNFSM4UHL444A>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&utm_campaign=notification-email>
> .
>

sloriot commented:
Warnings in https://cgal.geometryfactory.com/CGAL/testsuite/results-5.4-Ic-30.shtml:
- Arrangement_on_surface_2_Examples
- BGL_arrangement_2_Examples
- Boolean_set_operations_2
- Envelope_2_Examples
- Envelope_3_Examples
- Minkowski_sum_2
- Minkowski_sum_2_Examples
- Surface_sweep_2
- Surface_sweep_2_Examples
- Visibility_2
- Visibility_2_Examples

sloriot commented:
You might also want to check if some of those doc warnings are legitimate: https://cgal.geometryfactory.com/CGAL/Manual_doxygen_test/CGAL-5.4-Ic-30/logs_master/Arrangement_on_surface_2.log

(as it's with doxygen master, it might be that they are just doxygen "bugs")

efifogel commented:
I believe that I've fixed these
   ____  _        ____             _
  /_____/_) o    /__________  __  //
 (____ (   (    (    (_/ (_/-(-'_(/
                         _/




On Thu, 12 Aug 2021 at 13:08, Maxime GIMENO ***@***.***>
wrote:

> There are still warnings and errors, but only in Arrangement examples:
> https://cgal.geometryfactory.com/~mgimeno/test_suite/TESTRESULTS/index.shtml#Arrangement_on_surface_2_Examples
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/CGAL/cgal/pull/5223#issuecomment-897512655>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABVBNOH6DJGHSYGGNXDE7JTT4OMRPANCNFSM4UHL444A>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&utm_campaign=notification-email>
> .
>

efifogel commented:
I mean, I believe that I've fixed the warnings in the code.
As far as Doxygen goes, I couldn't find any problem, except perhaps for 1
or 2, which I fixed.
1. The unmatching /span comes from nowhere---I've removed all occurrences
of span and I still get it.
2. The unresolved label seems to come from referencing before definition,
but this should be ok.
   ____  _        ____             _
  /_____/_) o    /__________  __  //
 (____ (   (    (    (_/ (_/-(-'_(/
                         _/




On Thu, 19 Aug 2021 at 13:29, Efi Fogel ***@***.***> wrote:

> I believe that I've fixed these
>    ____  _        ____             _
>   /_____/_) o    /__________  __  //
>  (____ (   (    (    (_/ (_/-(-'_(/
>                          _/
>
>
>
>
> On Thu, 12 Aug 2021 at 13:08, Maxime GIMENO ***@***.***>
> wrote:
>
>> There are still warnings and errors, but only in Arrangement examples:
>> https://cgal.geometryfactory.com/~mgimeno/test_suite/TESTRESULTS/index.shtml#Arrangement_on_surface_2_Examples
>>
>> —
>> You are receiving this because you were mentioned.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/CGAL/cgal/pull/5223#issuecomment-897512655>, or
>> unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/ABVBNOH6DJGHSYGGNXDE7JTT4OMRPANCNFSM4UHL444A>
>> .
>> Triage notifications on the go with GitHub Mobile for iOS
>> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
>> or Android
>> <https://play.google.com/store/apps/details?id=com.github.android&utm_campaign=notification-email>
>> .
>>
>

lrineau commented on Arrangement_on_surface_2/examples/Arrangement_on_surface_2/bgl_dual_adapter.cpp:
```diff
@@ -1,10 +1,11 @@
 //! \file examples/Arrangement_on_surface_2/bgl_dual_adapter.cpp
 // Adapting the dual of an arrangement to a BGL graph.
 
+#include <CGAL/basic.h>
```
```suggestion
#include <CGAL/config.h>
```
In my opinion `<CGAL/basic.h>` is an evil thing that includes too much of CGAL. Here `<CGAL/config.h>` should be enough.

efifogel commented:
Done
   ____  _        ____             _
  /_____/_) o    /__________  __  //
 (____ (   (    (    (_/ (_/-(-'_(/
                         _/




On Thu, 19 Aug 2021 at 16:33, Laurent Rineau ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In
> Arrangement_on_surface_2/examples/Arrangement_on_surface_2/bgl_dual_adapter.cpp
> <https://github.com/CGAL/cgal/pull/5223#discussion_r692114373>:
>
> > @@ -1,10 +1,11 @@
>
>  //! \file examples/Arrangement_on_surface_2/bgl_dual_adapter.cpp
>
>  // Adapting the dual of an arrangement to a BGL graph.
>
>
>
> +#include <CGAL/basic.h>
>
>
> ⬇️ Suggested change
>
> -#include <CGAL/basic.h>
>
> +#include <CGAL/config.h>
>
>
> In my opinion <CGAL/basic.h> is an evil thing that includes too much of
> CGAL. Here <CGAL/config.h> should be enough.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/CGAL/cgal/pull/5223#pullrequestreview-733993240>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABVBNOCTC57Z4SEYWTLPIMTT5UBZRANCNFSM4UHL444A>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&utm_campaign=notification-email>
> .
>

sloriot commented:
Almost all fixed!

Errors:
- https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-5.4-Ic-33/Arrangement_on_surface_2_Examples/TestReport_afabri_x64_Cygwin-Windows10_MSVC2017-Release-64bits.gz
- https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-5.4-Ic-33/Arrangement_on_surface_2_Examples/TestReport_Sosno_MSVC-2019-Community-Release.gz

warnings:
- https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-5.4-Ic-33/Arrangement_on_surface_2_Examples/TestReport_lrineau_Ubuntu-latest-GCC6-CXX1z.gz
- https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-5.4-Ic-33/Arrangement_on_surface_2_Examples/TestReport_afabri_x64_Cygwin-Windows10_MSVC2017-Debug-64bits.gz

efifogel commented:
I've fixed all except for one warning, which is actually caused by a
deficiency of boost.
See constructor of vector_property_map (in
<boost/property_map/vector_property_map.hpp>)
It accepts 'unsigned' but it should accept 'size_t'.
So, I hope this was the last one.
   ____  _        ____             _
  /_____/_) o    /__________  __  //
 (____ (   (    (    (_/ (_/-(-'_(/
                         _/




On Mon, 23 Aug 2021 at 12:56, Sebastien Loriot ***@***.***>
wrote:

> Almost all fixed!
>
> Errors:
>
>    -
>    https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-5.4-Ic-33/Arrangement_on_surface_2_Examples/TestReport_afabri_x64_Cygwin-Windows10_MSVC2017-Release-64bits.gz
>    -
>    https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-5.4-Ic-33/Arrangement_on_surface_2_Examples/TestReport_Sosno_MSVC-2019-Community-Release.gz
>
> warnings:
>
>    -
>    https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-5.4-Ic-33/Arrangement_on_surface_2_Examples/TestReport_lrineau_Ubuntu-latest-GCC6-CXX1z.gz
>    -
>    https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-5.4-Ic-33/Arrangement_on_surface_2_Examples/TestReport_afabri_x64_Cygwin-Windows10_MSVC2017-Debug-64bits.gz
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/CGAL/cgal/pull/5223#issuecomment-903618447>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABVBNOBA6F5P5ACQ3WBREZTT6ILNBANCNFSM4UHL444A>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&utm_campaign=notification-email>
> .
>

sloriot commented:
Arrangement_examples are now fully red.

efifogel commented:
Ops, forgot to push one file.
Now it's pushed.
Let's try again, sorry.
   ____  _        ____             _
  /_____/_) o    /__________  __  //
 (____ (   (    (    (_/ (_/-(-'_(/
                         _/




On Tue, 24 Aug 2021 at 10:15, Sebastien Loriot ***@***.***>
wrote:

> Arrangement_examples are now fully red.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/CGAL/cgal/pull/5223#issuecomment-904385242>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABVBNOGRCOGWUA2OWBVP4I3T6NBH7ANCNFSM4UHL444A>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&utm_campaign=notification-email>
> .
>

sloriot commented:
Warnings left (exhausive list):
 - `CGAL-5.4-Ic-36/Arrangement_on_surface_2_Examples/TestReport_afabri_x64_Cygwin-Windows10_MSVC2017-Debug-64bits.gz:C:\CGAL\test\CGAL-5.4-Ic-36\cmake\platforms\x64_Cygwin-Windows10_MSVC2017-Debug-64bits\test\Arrangement_on_surface_2_Examples\bgl_primal_adapter.cpp(38): warning C4267: 'argument': conversion from 'size_t' to 'unsigned int', possible loss of data`
 - `CGAL-5.4-Ic-36/Arrangement_on_surface_2_Examples/TestReport_afabri_x64_Cygwin-Windows10_MSVC2017-Release-64bits.gz:C:\CGAL\test\CGAL-5.4-Ic-36\cmake\platforms\x64_Cygwin-Windows10_MSVC2017-Release-64bits\test\Arrangement_on_surface_2_Examples\bgl_primal_adapter.cpp(38): warning C4267: 'argument': conversion from 'size_t' to 'unsigned int', possible loss of data`
 - `CGAL-5.4-Ic-36/Arrangement_on_surface_2_Examples/TestReport_afabri_x64_Cygwin-Windows10_MSVC2017-Release-64bits.gz:C:\CGAL\test\CGAL-5.4-Ic-36\include\CGAL/Arr_polycurve_basic_traits_2.h(1698): warning C4267: 'initializing': conversion from 'size_t' to 'unsigned int', possible loss of data`
 - `CGAL-5.4-Ic-36/Arrangement_on_surface_2_Examples/TestReport_afabri_x64_Cygwin-Windows10_MSVC2017-Release-64bits.gz:C:\CGAL\test\CGAL-5.4-Ic-36\include\CGAL/Arr_polycurve_basic_traits_2.h(1699): warning C4267: 'initializing': conversion from 'size_t' to 'unsigned int', possible loss of data`
 - `CGAL-5.4-Ic-36/Arrangement_on_surface_2_Examples/TestReport_lrineau_Fedora-rawhide-Release.gz:/mnt/testsuite/include/CGAL/Curved_kernel_via_analysis_2/Point_2.h:55:7: warning: ‘<unnamed>.CGAL::internal::Point_2_rep<CGAL::Curved_kernel_via_analysis_2<CGAL::Algebraic_curve_kernel_2<CGAL::Algebraic_kernel_d_1<CORE::BigInt, CORE::BigRat, CGAL::internal::Algebraic_real_rep<CORE::BigInt, CORE::BigRat>, CGAL::internal::Descartes<CGAL::Polynomial<CORE::BigInt>, CORE::BigRat> > > > >::_m_location’ is used uninitialized [-Wuninitialized]`
 - `CGAL-5.4-Ic-36/Arrangement_on_surface_2_Examples/TestReport_lrineau_Fedora-rawhide-Release.gz:/mnt/testsuite/include/CGAL/Curved_kernel_via_analysis_2/Point_2.h:55:7: warning: ‘<unnamed>.CGAL::internal::Point_2_rep<CGAL::Curved_kernel_via_analysis_2<CGAL::Algebraic_curve_kernel_2<CGAL::Algebraic_kernel_d_1<CORE::BigInt, CORE::BigRat, CGAL::internal::Algebraic_real_rep<CORE::BigInt, CORE::BigRat>, CGAL::internal::Descartes<CGAL::Polynomial<CORE::BigInt>, CORE::BigRat> > > > >::_m_location’ may be used uninitialized [-Wmaybe-uninitialized]`
 - `CGAL-5.4-Ic-36/Arrangement_on_surface_2_Examples/TestReport_lrineau_Ubuntu-GCC_master_CXX20-Release.gz:/mnt/testsuite/include/CGAL/Algebraic_kernel_d/Curve_analysis_2.h:192:12: warning: '*(const __vector(4) int*)((char*)&curr_event + offsetof(CGAL::internal::Curve_analysis_2_rep<CGAL::Algebraic_curve_kernel_2<CGAL::Algebraic_kernel_d_1<CORE::BigInt, CORE::BigRat, CGAL::internal::Algebraic_real_rep<CORE::BigInt, CORE::BigRat>, CGAL::internal::Descartes<CGAL::Polynomial<CORE::BigInt>, CORE::BigRat> > > >::Event_coordinate_1, CGAL::internal::Curve_analysis_2_rep<CGAL::Algebraic_curve_kernel_2<CGAL::Algebraic_kernel_d_1<CORE::BigInt, CORE::BigRat, CGAL::internal::Algebraic_real_rep<CORE::BigInt, CORE::BigRat>, CGAL::internal::Descartes<CGAL::Polynomial<CORE::BigInt>, CORE::BigRat> > > >::Event_coordinate_1::mult_of_prim_res_root))' may be used uninitialized [-Wmaybe-uninitialized]`
 - `CGAL-5.4-Ic-36/Arrangement_on_surface_2_Examples/TestReport_lrineau_Ubuntu-GCC_master_CXX20-Release.gz:/mnt/testsuite/include/CGAL/Algebraic_kernel_d/Curve_analysis_2.h:192:12: warning: '*(long unsigned int*)((char*)&curr_event + offsetof(CGAL::internal::Curve_analysis_2_rep<CGAL::Algebraic_curve_kernel_2<CGAL::Algebraic_kernel_d_1<CORE::BigInt, CORE::BigRat, CGAL::internal::Algebraic_real_rep<CORE::BigInt, CORE::BigRat>, CGAL::internal::Descartes<CGAL::Polynomial<CORE::BigInt>, CORE::BigRat> > > >::Event_coordinate_1, CGAL::internal::Curve_analysis_2_rep<CGAL::Algebraic_curve_kernel_2<CGAL::Algebraic_kernel_d_1<CORE::BigInt, CORE::BigRat, CGAL::internal::Algebraic_real_rep<CORE::BigInt, CORE::BigRat>, CGAL::internal::Descartes<CGAL::Polynomial<CORE::BigInt>, CORE::BigRat> > > >::Event_coordinate_1::mult_of_prim_lcoeff_root))' may be used uninitialized [-Wmaybe-uninitialized]`
 - `CGAL-5.4-Ic-36/Arrangement_on_surface_2_Examples/TestReport_lrineau_Ubuntu-GCC_master_CXX20-Release.gz:/mnt/testsuite/include/CGAL/Arr_geodesic_arc_on_sphere_traits_2.h:1488:51: warning: unused parameter 'p' [-Wunused-parameter]`
 - `CGAL-5.4-Ic-36/Arrangement_on_surface_2_Examples/TestReport_lrineau_Ubuntu-GCC_master_CXX20-Release.gz:/mnt/testsuite/include/CGAL/Curved_kernel_via_analysis_2/Point_2.h:55:7: warning: '<unnamed>.CGAL::internal::Point_2_rep<CGAL::Curved_kernel_via_analysis_2<CGAL::Algebraic_curve_kernel_2<CGAL::Algebraic_kernel_d_1<CORE::BigInt, CORE::BigRat, CGAL::internal::Algebraic_real_rep<CORE::BigInt, CORE::BigRat>, CGAL::internal::Descartes<CGAL::Polynomial<CORE::BigInt>, CORE::BigRat> > > > >::_m_location' is used uninitialized [-Wuninitialized]`
 - `CGAL-5.4-Ic-36/Arrangement_on_surface_2_Examples/TestReport_lrineau_Ubuntu-GCC_master_CXX20-Release.gz:/mnt/testsuite/include/CGAL/Curved_kernel_via_analysis_2/Point_2.h:55:7: warning: '<unnamed>.CGAL::internal::Point_2_rep<CGAL::Curved_kernel_via_analysis_2<CGAL::Algebraic_curve_kernel_2<CGAL::Algebraic_kernel_d_1<CORE::BigInt, CORE::BigRat, CGAL::internal::Algebraic_real_rep<CORE::BigInt, CORE::BigRat>, CGAL::internal::Descartes<CGAL::Polynomial<CORE::BigInt>, CORE::BigRat> > > > >::_m_location' may be used uninitialized [-Wmaybe-uninitialized]`
 - `CGAL-5.4-Ic-36/Arrangement_on_surface_2_Examples/TestReport_lrineau_Ubuntu-GCC_master_CXX20-Release.gz:/usr/local/boost/include/boost/optional/optional.hpp:751:58: warning: implicitly-declared 'CGAL::internal::Bitstream_descartes<CGAL::internal::Bitstream_descartes_rndl_tree_traits<CGAL::internal::Bitstream_coefficient_kernel_at_alpha<CGAL::Algebraic_curve_kernel_2<CGAL::Algebraic_kernel_d_1<CORE::BigInt, CORE::BigRat, CGAL::internal::Algebraic_real_rep<CORE::BigInt, CORE::BigRat>, CGAL::internal::Descartes<CGAL::Polynomial<CORE::BigInt>, CORE::BigRat> > > > > >& CGAL::internal::Bitstream_descartes<CGAL::internal::Bitstream_descartes_rndl_tree_traits<CGAL::internal::Bitstream_coefficient_kernel_at_alpha<CGAL::Algebraic_curve_kernel_2<CGAL::Algebraic_kernel_d_1<CORE::BigInt, CORE::BigRat, CGAL::internal::Algebraic_real_rep<CORE::BigInt, CORE::BigRat>, CGAL::internal::Descartes<CGAL::Polynomial<CORE::BigInt>, CORE::BigRat> > > > > >::operator=(const CGAL::internal::Bitstream_descartes<CGAL::internal::Bitstream_descartes_rndl_tree_traits<CGAL::internal::Bitstream_coefficient_kernel_at_alpha<CGAL::Algebraic_curve_kernel_2<CGAL::Algebraic_kernel_d_1<CORE::BigInt, CORE::BigRat, CGAL::internal::Algebraic_real_rep<CORE::BigInt, CORE::BigRat>, CGAL::internal::Descartes<CGAL::Polynomial<CORE::BigInt>, CORE::BigRat> > > > > >&)' is deprecated [-Wdeprecated-copy]`
 - `CGAL-5.4-Ic-36/Arrangement_on_surface_2_Examples/TestReport_lrineau_Ubuntu-GCC_master_CXX20-Release.gz:/usr/local/boost/include/boost/tuple/detail/tuple_basic.hpp:591:16: warning: implicitly-declared 'constexpr boost::iterators::counting_iterator<CGAL::Arrangement_on_surface_2<CGAL::Arr_segment_traits_2<CGAL::Epeck>, CGAL::Arr_bounded_planar_topology_traits_2<CGAL::Arr_segment_traits_2<CGAL::Epeck>, CGAL::Arr_default_dcel<CGAL::Arr_segment_traits_2<CGAL::Epeck> > > >::_Valid_vertex_iterator, boost::use_default, boost::use_default>& boost::iterators::counting_iterator<CGAL::Arrangement_on_surface_2<CGAL::Arr_segment_traits_2<CGAL::Epeck>, CGAL::Arr_bounded_planar_topology_traits_2<CGAL::Arr_segment_traits_2<CGAL::Epeck>, CGAL::Arr_default_dcel<CGAL::Arr_segment_traits_2<CGAL::Epeck> > > >::_Valid_vertex_iterator, boost::use_default, boost::use_default>::operator=(const boost::iterators::counting_iterator<CGAL::Arrangement_on_surface_2<CGAL::Arr_segment_traits_2<CGAL::Epeck>, CGAL::Arr_bounded_planar_topology_traits_2<CGAL::Arr_segment_traits_2<CGAL::Epeck>, CGAL::Arr_default_dcel<CGAL::Arr_segment_traits_2<CGAL::Epeck> > > >::_Valid_vertex_iterator, boost::use_default, boost::use_default>&)' is deprecated [-Wdeprecated-copy]`
 - `CGAL-5.4-Ic-36/Arrangement_on_surface_2_Examples/TestReport_lrineau_Ubuntu-GCC_master_CXX20-Release.gz:/usr/local/boost/include/boost/tuple/detail/tuple_basic.hpp:591:16: warning: implicitly-declared 'constexpr boost::iterators::counting_iterator<CGAL::I_Filtered_iterator<CGAL::internal::CC_iterator<CGAL::Compact_container<CGAL::Arr_face<CGAL::Arr_vertex_base<CGAL::Point_2<CGAL::Epeck> >, CGAL::Arr_halfedge_base<CGAL::Arr_segment_2<CGAL::Epeck> >, CGAL::Arr_extended_face<CGAL::Arr_face_base, unsigned int> >, boost::fast_pool_allocator<CGAL::Arr_face<CGAL::Arr_vertex_base<CGAL::Point_2<CGAL::Epeck> >, CGAL::Arr_halfedge_base<CGAL::Arr_segment_2<CGAL::Epeck> >, CGAL::Arr_extended_face<CGAL::Arr_face_base, unsigned int> >, boost::default_user_allocator_new_delete, std::mutex, 32, 0>, CGAL::Default, CGAL::Default>, false>, CGAL::Arrangement_on_surface_2<CGAL::Arr_segment_traits_2<CGAL::Epeck>, CGAL::Arr_bounded_planar_topology_traits_2<CGAL::Arr_segment_traits_2<CGAL::Epeck>, CGAL::Arr_face_extended_dcel<CGAL::Arr_segment_traits_2<CGAL::Epeck>, unsigned int> > >::_Is_valid_face, CGAL::Arrangement_on_surface_2<CGAL::Arr_segment_traits_2<CGAL::Epeck>, CGAL::Arr_bounded_planar_topology_traits_2<CGAL::Arr_segment_traits_2<CGAL::Epeck>, CGAL::Arr_face_extended_dcel<CGAL::Arr_segment_traits_2<CGAL::Epeck>, unsigned int> > >::Face, long int, std::bidirectional_iterator_tag>, boost::use_default, boost::use_default>& boost::iterators::counting_iterator<CGAL::I_Filtered_iterator<CGAL::internal::CC_iterator<CGAL::Compact_container<CGAL::Arr_face<CGAL::Arr_vertex_base<CGAL::Point_2<CGAL::Epeck> >, CGAL::Arr_halfedge_base<CGAL::Arr_segment_2<CGAL::Epeck> >, CGAL::Arr_extended_face<CGAL::Arr_face_base, unsigned int> >, boost::fast_pool_allocator<CGAL::Arr_face<CGAL::Arr_vertex_base<CGAL::Point_2<CGAL::Epeck> >, CGAL::Arr_halfedge_base<CGAL::Arr_segment_2<CGAL::Epeck> >, CGAL::Arr_extended_face<CGAL::Arr_face_base, unsigned int> >, boost::default_user_allocator_new_delete, std::mutex, 32, 0>, CGAL::Default, CGAL::Default>, false>, CGAL::Arrangement_on_surface_2<CGAL::Arr_segment_traits_2<CGAL::Epeck>, CGAL::Arr_bounded_planar_topology_traits_2<CGAL::Arr_segment_traits_2<CGAL::Epeck>, CGAL::Arr_face_extended_dcel<CGAL::Arr_segment_traits_2<CGAL::Epeck>, unsigned int> > >::_Is_valid_face, CGAL::Arrangement_on_surface_2<CGAL::Arr_segment_traits_2<CGAL::Epeck>, CGAL::Arr_bounded_planar_topology_traits_2<CGAL::Arr_segment_traits_2<CGAL::Epeck>, CGAL::Arr_face_extended_dcel<CGAL::Arr_segment_traits_2<CGAL::Epeck>, unsigned int> > >::Face, long int, std::bidirectional_iterator_tag>, boost::use_default, boost::use_default>::operator=(const boost::iterators::counting_iterator<CGAL::I_Filtered_iterator<CGAL::internal::CC_iterator<CGAL::Compact_container<CGAL::Arr_face<CGAL::Arr_vertex_base<CGAL::Point_2<CGAL::Epeck> >, CGAL::Arr_halfedge_base<CGAL::Arr_segment_2<CGAL::Epeck> >, CGAL::Arr_extended_face<CGAL::Arr_face_base, unsigned int> >, boost::fast_pool_allocator<CGAL::Arr_face<CGAL::Arr_vertex_base<CGAL::Point_2<CGAL::Epeck> >, CGAL::Arr_halfedge_base<CGAL::Arr_segment_2<CGAL::Epeck> >, CGAL::Arr_extended_face<CGAL::Arr_face_base, unsigned int> >, boost::default_user_allocator_new_delete, std::mutex, 32, 0>, CGAL::Default, CGAL::Default>, false>, CGAL::Arrangement_on_surface_2<CGAL::Arr_segment_traits_2<CGAL::Epeck>, CGAL::Arr_bounded_planar_topology_traits_2<CGAL::Arr_segment_traits_2<CGAL::Epeck>, CGAL::Arr_face_extended_dcel<CGAL::Arr_segment_traits_2<CGAL::Epeck>, unsigned int> > >::_Is_valid_face, CGAL::Arrangement_on_surface_2<CGAL::Arr_segment_traits_2<CGAL::Epeck>, CGAL::Arr_bounded_planar_topology_traits_2<CGAL::Arr_segment_traits_2<CGAL::Epeck>, CGAL::Arr_face_extended_dcel<CGAL::Arr_segment_traits_2<CGAL::Epeck>, unsigned int> > >::Face, long int, std::bidirectional_iterator_tag>, boost::use_default, boost::use_default>&)' is deprecated [-Wdeprecated-copy]`
 - `CGAL-5.4-Ic-36/Arrangement_on_surface_2_Examples/TestReport_lrineau_Ubuntu-GCC_master_CXX20-Release.gz:/usr/local/boost/include/boost/tuple/detail/tuple_basic.hpp:592:21: warning: implicitly-declared 'constexpr boost::iterators::counting_iterator<CGAL::Arrangement_on_surface_2<CGAL::Arr_segment_traits_2<CGAL::Epeck>, CGAL::Arr_bounded_planar_topology_traits_2<CGAL::Arr_segment_traits_2<CGAL::Epeck>, CGAL::Arr_default_dcel<CGAL::Arr_segment_traits_2<CGAL::Epeck> > > >::_Valid_vertex_iterator, boost::use_default, boost::use_default>& boost::iterators::counting_iterator<CGAL::Arrangement_on_surface_2<CGAL::Arr_segment_traits_2<CGAL::Epeck>, CGAL::Arr_bounded_planar_topology_traits_2<CGAL::Arr_segment_traits_2<CGAL::Epeck>, CGAL::Arr_default_dcel<CGAL::Arr_segment_traits_2<CGAL::Epeck> > > >::_Valid_vertex_iterator, boost::use_default, boost::use_default>::operator=(const boost::iterators::counting_iterator<CGAL::Arrangement_on_surface_2<CGAL::Arr_segment_traits_2<CGAL::Epeck>, CGAL::Arr_bounded_planar_topology_traits_2<CGAL::Arr_segment_traits_2<CGAL::Epeck>, CGAL::Arr_default_dcel<CGAL::Arr_segment_traits_2<CGAL::Epeck> > > >::_Valid_vertex_iterator, boost::use_default, boost::use_default>&)' is deprecated [-Wdeprecated-copy]`
 - `CGAL-5.4-Ic-36/Arrangement_on_surface_2_Examples/TestReport_lrineau_Ubuntu-GCC_master_CXX20-Release.gz:/usr/local/boost/include/boost/tuple/detail/tuple_basic.hpp:592:21: warning: implicitly-declared 'constexpr boost::iterators::counting_iterator<CGAL::I_Filtered_iterator<CGAL::internal::CC_iterator<CGAL::Compact_container<CGAL::Arr_face<CGAL::Arr_vertex_base<CGAL::Point_2<CGAL::Epeck> >, CGAL::Arr_halfedge_base<CGAL::Arr_segment_2<CGAL::Epeck> >, CGAL::Arr_extended_face<CGAL::Arr_face_base, unsigned int> >, boost::fast_pool_allocator<CGAL::Arr_face<CGAL::Arr_vertex_base<CGAL::Point_2<CGAL::Epeck> >, CGAL::Arr_halfedge_base<CGAL::Arr_segment_2<CGAL::Epeck> >, CGAL::Arr_extended_face<CGAL::Arr_face_base, unsigned int> >, boost::default_user_allocator_new_delete, std::mutex, 32, 0>, CGAL::Default, CGAL::Default>, false>, CGAL::Arrangement_on_surface_2<CGAL::Arr_segment_traits_2<CGAL::Epeck>, CGAL::Arr_bounded_planar_topology_traits_2<CGAL::Arr_segment_traits_2<CGAL::Epeck>, CGAL::Arr_face_extended_dcel<CGAL::Arr_segment_traits_2<CGAL::Epeck>, unsigned int> > >::_Is_valid_face, CGAL::Arrangement_on_surface_2<CGAL::Arr_segment_traits_2<CGAL::Epeck>, CGAL::Arr_bounded_planar_topology_traits_2<CGAL::Arr_segment_traits_2<CGAL::Epeck>, CGAL::Arr_face_extended_dcel<CGAL::Arr_segment_traits_2<CGAL::Epeck>, unsigned int> > >::Face, long int, std::bidirectional_iterator_tag>, boost::use_default, boost::use_default>& boost::iterators::counting_iterator<CGAL::I_Filtered_iterator<CGAL::internal::CC_iterator<CGAL::Compact_container<CGAL::Arr_face<CGAL::Arr_vertex_base<CGAL::Point_2<CGAL::Epeck> >, CGAL::Arr_halfedge_base<CGAL::Arr_segment_2<CGAL::Epeck> >, CGAL::Arr_extended_face<CGAL::Arr_face_base, unsigned int> >, boost::fast_pool_allocator<CGAL::Arr_face<CGAL::Arr_vertex_base<CGAL::Point_2<CGAL::Epeck> >, CGAL::Arr_halfedge_base<CGAL::Arr_segment_2<CGAL::Epeck> >, CGAL::Arr_extended_face<CGAL::Arr_face_base, unsigned int> >, boost::default_user_allocator_new_delete, std::mutex, 32, 0>, CGAL::Default, CGAL::Default>, false>, CGAL::Arrangement_on_surface_2<CGAL::Arr_segment_traits_2<CGAL::Epeck>, CGAL::Arr_bounded_planar_topology_traits_2<CGAL::Arr_segment_traits_2<CGAL::Epeck>, CGAL::Arr_face_extended_dcel<CGAL::Arr_segment_traits_2<CGAL::Epeck>, unsigned int> > >::_Is_valid_face, CGAL::Arrangement_on_surface_2<CGAL::Arr_segment_traits_2<CGAL::Epeck>, CGAL::Arr_bounded_planar_topology_traits_2<CGAL::Arr_segment_traits_2<CGAL::Epeck>, CGAL::Arr_face_extended_dcel<CGAL::Arr_segment_traits_2<CGAL::Epeck>, unsigned int> > >::Face, long int, std::bidirectional_iterator_tag>, boost::use_default, boost::use_default>::operator=(const boost::iterators::counting_iterator<CGAL::I_Filtered_iterator<CGAL::internal::CC_iterator<CGAL::Compact_container<CGAL::Arr_face<CGAL::Arr_vertex_base<CGAL::Point_2<CGAL::Epeck> >, CGAL::Arr_halfedge_base<CGAL::Arr_segment_2<CGAL::Epeck> >, CGAL::Arr_extended_face<CGAL::Arr_face_base, unsigned int> >, boost::fast_pool_allocator<CGAL::Arr_face<CGAL::Arr_vertex_base<CGAL::Point_2<CGAL::Epeck> >, CGAL::Arr_halfedge_base<CGAL::Arr_segment_2<CGAL::Epeck> >, CGAL::Arr_extended_face<CGAL::Arr_face_base, unsigned int> >, boost::default_user_allocator_new_delete, std::mutex, 32, 0>, CGAL::Default, CGAL::Default>, false>, CGAL::Arrangement_on_surface_2<CGAL::Arr_segment_traits_2<CGAL::Epeck>, CGAL::Arr_bounded_planar_topology_traits_2<CGAL::Arr_segment_traits_2<CGAL::Epeck>, CGAL::Arr_face_extended_dcel<CGAL::Arr_segment_traits_2<CGAL::Epeck>, unsigned int> > >::_Is_valid_face, CGAL::Arrangement_on_surface_2<CGAL::Arr_segment_traits_2<CGAL::Epeck>, CGAL::Arr_bounded_planar_topology_traits_2<CGAL::Arr_segment_traits_2<CGAL::Epeck>, CGAL::Arr_face_extended_dcel<CGAL::Arr_segment_traits_2<CGAL::Epeck>, unsigned int> > >::Face, long int, std::bidirectional_iterator_tag>, boost::use_default, boost::use_default>&)' is deprecated [-Wdeprecated-copy]`
 - `CGAL-5.4-Ic-36/Arrangement_on_surface_2_Examples/TestReport_lrineau_Ubuntu-GCC_master_CXX20-Release.gz:/usr/local/include/c++/12.0.0/bits/char_traits.h:355:56: warning: 'void* __builtin_memcpy(void*, const void*, long unsigned int)' specified bound between 18446744073709551602 and 18446744073709551615 exceeds maximum object size 9223372036854775807 [-Wstringop-overflow=]
 - `CGAL-5.4-Ic-36/Arrangement_on_surface_2_Examples/TestReport_monet_x86_64-apple_bigsur_clang1200-release-64bits.gz:/Users/monet/Documents/test-suite/clang-default/CGAL-5.4-Ic-36/test/Arrangement_on_surface_2_Examples/Extended_face_property_map.h:16:58: warning: unused parameter 'map' [-Wunused-parameter]`
 - `CGAL-5.4-Ic-36/Arrangement_on_surface_2_Examples/TestReport_Sosno_MSVC-2019-Community-Release.gz:    78>C:\dev\boost\boost_1_67_0\boost\graph\named_function_params.hpp(240): warning C4172: returning address of local variable or temporary`
 - `CGAL-5.4-Ic-36/Arrangement_on_surface_2_Examples/TestReport_Sosno_MSVC-2019-Community-Release.gz:    79>C:\dev\boost\boost_1_67_0\boost\graph\named_function_params.hpp(240): warning C4172: returning address of local variable or temporary`
 - `CGAL-5.4-Ic-36/Arrangement_on_surface_2_Examples/TestReport_Sosno_MSVC-2019-Community-Release.gz:         C:\dev\boost\boost_1_67_0\boost\graph\named_function_params.hpp(240): warning C4172: returning address of local variable or temporary`
 - `CGAL-5.4-Ic-36/Arrangement_on_surface_2_Examples/TestReport_Sosno_MSVC-2019-Community-Release.gz:         C:\dev\boost\boost_1_67_0\boost\graph\named_function_params.hpp(240): warning C4172: returning address of local variable or temporary`
 - `CGAL-5.4-Ic-36/Arrangement_on_surface_2_Examples/TestReport_Sosno_MSVC-2022-Preview-Release.gz:   109>C:\cgal_root\CGAL-5.4-Ic-36\include\CGAL/Arr_polycurve_basic_traits_2.h(1698,1): warning C4267: 'initializing': conversion from 'size_t' to 'unsigned int', possible loss of data`
 - `CGAL-5.4-Ic-36/Arrangement_on_surface_2_Examples/TestReport_Sosno_MSVC-2022-Preview-Release.gz:   109>C:\cgal_root\CGAL-5.4-Ic-36\include\CGAL/Arr_polycurve_basic_traits_2.h(1699,1): warning C4267: 'initializing': conversion from 'size_t' to 'unsigned int', possible loss of data`
 - `CGAL-5.4-Ic-36/Arrangement_on_surface_2_Examples/TestReport_Sosno_MSVC-2022-Preview-Release.gz:    75>C:\cgal_root\CGAL-5.4-Ic-36\test\Arrangement_on_surface_2_Examples\bgl_primal_adapter.cpp(38,49): warning C4267: 'argument': conversion from 'size_t' to 'unsigned int', possible loss of data`
 - `CGAL-5.4-Ic-36/Arrangement_on_surface_2_Examples/TestReport_Sosno_MSVC-2022-Preview-Release.gz:         C:\cgal_root\CGAL-5.4-Ic-36\include\CGAL/Arr_polycurve_basic_traits_2.h(1698,1): warning C4267: 'initializing': conversion from 'size_t' to 'unsigned int', possible loss of data`
 - `CGAL-5.4-Ic-36/Arrangement_on_surface_2_Examples/TestReport_Sosno_MSVC-2022-Preview-Release.gz:         C:\cgal_root\CGAL-5.4-Ic-36\include\CGAL/Arr_polycurve_basic_traits_2.h(1699,1): warning C4267: 'initializing': conversion from 'size_t' to 'unsigned int', possible loss of data`
 - `CGAL-5.4-Ic-36/Arrangement_on_surface_2_Examples/TestReport_Sosno_MSVC-2022-Preview-Release.gz:         C:\cgal_root\CGAL-5.4-Ic-36\test\Arrangement_on_surface_2_Examples\bgl_primal_adapter.cpp(38,49): warning C4267: 'argument': conversion from 'size_t' to 'unsigned int', possible loss of data`

maxGimeno commented:
Errors with LEDA : https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-5.4-Ic-41/Arrangement_on_surface_2/TestReport_lrineau_Fedora-with-LEDA.gz

sloriot commented:
Current branch size is 2.8MB. I could get it down to 2.2MB mainly by removing pdf files used for the latex version of the doc (not even sure it still works). @efifogel is it fine to not commit those files and keep only the png? The branch is in my remote (same name) and merged in integration.

efifogel commented:
Yes, no problem.
Remove those pd files.
I can generate them any time.
   ____  _        ____             _
  /_____/_) o    /__________  __  //
 (____ (   (    (    (_/ (_/-(-'_(/
                         _/




On Thu, 2 Sept 2021 at 20:15, Sebastien Loriot ***@***.***>
wrote:

> Current branch size is 2.8MB. I could get it down to 2.2MB mainly by
> removing pdf files used for the latex version of the doc (not even sure it
> still works). @efifogel <https://github.com/efifogel> is it fine to not
> commit those files and keep only the png? The branch is in my remote (same
> name) and merged in integration.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/CGAL/cgal/pull/5223#issuecomment-911895536>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABVBNOFXNGXT6FR5B6DRMITT76WKPANCNFSM4UHL444A>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>

sloriot commented:
Push-force-updated with my filtered version.

MaelRL commented:
Would it be worth a cgal.org news entry for `Arr_geodesic_arc_on_sphere_traits_2`?

efifogel commented:
If we generate such a news item, we can mention the movie clip from 2008:
http://acg.cs.tau.ac.il/projects/internal-projects/arr-geodesic-sphere/movie/aos-xvid.avi
In *Proceedings of the 24th ACM Symposium on Computational Geometry (SoCG)*,
pages 382-383, College Park, MD, USA, 2008.
   ____  _        ____             _
  /_____/_) o    /__________  __  //
 (____ (   (    (    (_/ (_/-(-'_(/
                         _/




On Wed, 10 Nov 2021 at 22:06, Mael ***@***.***> wrote:

> Would it be worth a cgal.org news entry for
> Arr_geodesic_arc_on_sphere_traits_2?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/CGAL/cgal/pull/5223#issuecomment-965702553>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABVBNODTHON5UQ5FITIXSGTULLGFVANCNFSM4UHL444A>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>
